src/axom/quest/io/STLWriter.cpp

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+#include "axom/quest/io/STLWriter.hpp"
+
+#include "axom/core/execution/reductions.hpp"
+#include "axom/mint/mesh/Mesh.hpp"
+#include "axom/mint/execution/interface.hpp"
+#include "axom/primal/geometry/Vector.hpp"
+
+#include <cstring>
+#include <fstream>
+
+namespace axom
+{
+namespace quest
+{
+namespace internal
+{
+
+/*!
+ * \brief Write a triangle to an output stream.
+ *
+ * \param out The stream to use for writing.
+ * \param binary Whether to write the triangle in binary format. The file must
+ *               have been opened in binary mode.
+ * \param coords The triangle coordinates.
+ * \param N The triangle normal.
+ */
+template <typename NormalType>
+void writeTriangle(std::ofstream &out, bool binary, double coords[3][3], const NormalType &N)
+{
+  if(binary)
+  {
+    using float32 = float;
+    using uint16 = unsigned short;
+    float32 n32[3], coords32[3][3];
+    for(int comp = 0; comp < 3; comp++)
+    {
+      n32[comp] = static_cast<float32>(N[comp]);
+      coords32[0][comp] = static_cast<float32>(coords[0][comp]);
```

Comment:
Aside: I didn't appreciate until now that binary STLs use floats for the geometry and normals